packages/language-server/README.md

```diff
@@ -0,0 +1,106 @@
+# `@mdx-js/language-service`
+
+## What is this?
+
+This package provides a [language server][lsp] for [MDX][].
+The language server provides intellisense based on [TypeScript][], as well as
+some markdown specific features.
+
+## When should I use this?
+
+You can use this package if you want to enhance your editor with intellisense
+for [MDX][] files.
+Some editors can consume this package directly, others need a plugin in order to
+consume this package.
+
+## Install
+
+This package is not published yet.
+
+## Use
+
+Usage of this package depends on your editor integration.
+
+This package provides the CLI `mdx-language-server`.
+Because this is based on [`vscode-languageserver`][vscode-languageserver], the
+same transports are supported.
+
+## Language server features
+
+This language server supports the following features:
+
+*   Markdown definitions
+*   Markdown hover hints
+*   TypeScript completions
+*   TypeScript definitions
+*   TypeScript diagnostics
+*   TypeScript hover hints
+*   TypeScript references
+
+[MDX][] doesn’t support TypeScript syntax, but it does support
+[types in JSDoc][jsdoc].
+The special type `Props` is used to determine the type used for `props`.
+For example:
+
+<!-- prettier-ignore -->
+
+```mdx
+{/**
+  * @type {object} Props
+  * @property {string} name
+  * Who to greet.
+  */}
```

Comment:
```suggestion
{/**
  * @typedef Props
  * @property {string} name
  *   Who to greet.
  */}
```

Shouldn’t this be a `typedef`? In which case, the `{...}` part is unneeded if `property`s are there?